src/main/scala/Column.scala

```diff
@@ -1034,13 +1096,88 @@ case class Column private (
     }
   }
 
+  def validateFormat(value: String, rowNumber: Long): Boolean = {
```

Comment:
Don't we usually only define `ErrorWithoutContext` in this class and then provide the context in the calling function?

Reply:
We also tend to pass errors/warnings back to the caller. I'm not sure I like the column holding its own errors/warnings. It'd be nice to be consistent in how we handle errors - i.e. continue to pass them back up.